### PR TITLE
Responsive styles component

### PR DESCRIPTION
**Pull request by toboid**
Extracts responsive-styles to be a separate package and updates author-head to consume it. Will be used next by author-profile.

### CODE REVIEW

File: packages/responsive-styles/__tests__/web/responsive-styles.test.js [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+  });
+  const tree = renderer.create(<BlueHello>Foo</BlueHello>).toJSON();
+
+  expect(tree).toMatchSnapshot();
+});
+
+it("supports responsive styles based on props", () => {
+  const ResponsiveHello = withResponsiveStyles(Hello, {
+    mediumUp: props => `
+      color: ${props.special ? "red;" : "blue"};
```
Comment (killanaca) [inline packages/responsive-styles/__tests__/web/responsive-styles.test.js]: just a missing semicolon

Comment (craigbilner) [reply packages/responsive-styles/__tests__/web/responsive-styles.test.js]: #needtofixprettier

Comment (toboid) [reply packages/responsive-styles/__tests__/web/responsive-styles.test.js]: thanks, updated

File: packages/responsive-styles/responsive-styles.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
 import styled from "styled-components";
 
+// Turn off prettier for this as the identified changes break this component
+// prettier-ignore
```
Comment (craigbilner) [inline packages/responsive-styles/responsive-styles.js]: @toboid minor: probably would have used the config file for consistency

**Comment by toboid** [comment]
@craigbilner updated so that this is now standalone and based on master. I can add the author-head changes to [#434](https://github.com/newsuk/times-components/pull/434) or submit them as a separate PR

**Pull request merged**